optaplanner-examples/src/main/java/org/optaplanner/examples/nurserostering/optional/score/NurseRosteringConstraintProvider.java

```diff
@@ -0,0 +1,480 @@
+                .join(ShiftAssignment.class, Joiners.equal((contract, employee) -> employee, ShiftAssignment::getEmployee),
+                        Joiners.filtering((c, e, s) -> s.isWeekend()))
+                .groupBy((contract, employee, shift) -> ImmutablePair.of(employee, contract),
+                        ExperimentalConstraintCollectors.consecutive((contract, employee, shift) -> shift.getShiftDate(),
+                                ShiftDate::getWeekendSundayIndex))
+                .flattenLast(ConsecutiveData::getConsecutiveSequences);
+    }
+
+    @Override
+    public Constraint[] defineConstraints(ConstraintFactory constraintFactory) {
+        return new Constraint[] {
+                oneShiftPerDay(constraintFactory),
+                minimumAndMaximumNumberOfAssignments(constraintFactory),
+                minimumNumberOfAssignmentsNoAssignments(constraintFactory),
+                minimumConsecutiveWorkingDays(constraintFactory),
+                maximumConsecutiveWorkingDays(constraintFactory),
+                minimumConsecutiveFreeDays(constraintFactory),
+                maximumConsecutiveFreeDays(constraintFactory),
+                maximumConsecutiveFreeDaysNoAssignments(constraintFactory),
+                consecutiveFreeDaysFirstBreak(constraintFactory),
+                consecutiveFreeDaysFinalBreak(constraintFactory),
+                minimumConsecutiveWorkingWeekends(constraintFactory),
+                maximumConsecutiveWorkingWeekends(constraintFactory),
+                startOnNotFirstDayOfWeekend(constraintFactory),
+                endOnNotLastDayOfWeekend(constraintFactory),
+                identicalShiftTypesDuringWeekend(constraintFactory),
+                dayOffRequest(constraintFactory),
+                dayOnRequest(constraintFactory),
+                shiftOffRequest(constraintFactory),
+                shiftOnRequest(constraintFactory),
+                alternativeSkill(constraintFactory),
+                unwantedPatternFreeBefore2DaysWithAWorkDayPattern(constraintFactory),
+                unwantedPatternShiftType2DaysPattern(constraintFactory),
+                unwantedPatternShiftType3DaysPattern(constraintFactory),
+        };
+    }
+
+    // ############################################################################
+    // Hard constraints
+    // ############################################################################
+
+    // A nurse can only work one shift per day, i.e. no two shift can be assigned to the same nurse on a day.
+    Constraint oneShiftPerDay(ConstraintFactory constraintFactory) {
+        return constraintFactory
+                .fromUniquePair(ShiftAssignment.class,
+                        Joiners.equal(ShiftAssignment::getEmployee),
+                        Joiners.equal(ShiftAssignment::getShiftDate))
+                .penalize("org.optaplanner.examples.nurserostering.solver", "oneShiftPerDay", HardSoftScore.ONE_HARD);
```

Comment:
I understand that you're including the package so that FULL_ASSERT works when compared against the DRL.
That said, this does not make the constraint provider easier to understand.
I'd rather change the DRL package than introduce this boilerplate in the constraint provider.

Reply:
I don't think it needed so FULL_ASSERT would work. Rather, it removes extra constraint matches shown in the score corruption overview when a score corruption does occur. I'll remove them.